lib/identity_cache/memoized_cache_proxy.rb

```diff
@@ -30,34 +30,47 @@ def with_memoization(&block)
     end
 
     def write(key, value)
-      memoized_key_values[key] = value if memoizing?
-      @cache_fetcher.write(key, value)
+      memoizing = memoizing?
+      ActiveSupport::Notifications.instrument('identity_cache.backend.write', memoizing: memoizing) do
+        memoized_key_values[key] = value if memoizing
+        @cache_fetcher.write(key, value)
+      end
     end
 
     def delete(key)
-      memoized_key_values.delete(key) if memoizing?
-      result = @cache_fetcher.delete(key)
-      IdentityCache.logger.debug { "[IdentityCache] delete #{ result ? 'recorded'  : 'failed'  } for #{key}" }
-      result
+      memoizing = memoizing?
+      ActiveSupport::Notifications.instrument('identity_cache.backend.delete', memoizing: memoizing) do
+        memoized_key_values.delete(key) if memoizing
+        result = @cache_fetcher.delete(key)
+        IdentityCache.logger.debug {"[IdentityCache] delete #{ result ? 'recorded' : 'failed'  } for #{key}"}
+        result
+      end
     end
 
     def fetch(key)
       used_cache_backend = true
       missed = false
-      value = if memoizing?
-        used_cache_backend = false
-        memoized_key_values.fetch(key) do
-          used_cache_backend = true
-          memoized_key_values[key] = @cache_fetcher.fetch(key) do
+      memoizing = memoizing?
+      value = ActiveSupport::Notifications.instrument('identity_cache.backend.fetch', { memoizing: memoizing }) do |payload|
```

Comment:
Having slept over this... I think, I'm voting for measuring one huge aggregate and publishing individual "times" (along side the current "per tier" metrics) as part of the payload. 
The reasoning being that I'd have to measure the `yield` block's execution time anyways, in the case of a miss (and have the caller subtract it himself, if there would be a `@cache_fetcher.fetch` notification); so I think we're better off _not_ throwing that piece of information away and just publish a notification, with the additional information of how time was spent:
 * hitting the memoizing layer
 * hitting memcached 
 * hitting the actual SoR

Given the API of the AS::Notification and the patterns used by IDC, I think that makes the most sense. wdyt?